File: java/563.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    int tileSum = 0;
    public int findTilt(TreeNode root) {
        dfs(root);
        return tileSum;
    }
    
    public int dfs(TreeNode root){
        Map<Integer,Integer> map = new HashMap<Integer,Integer>();
        if(root == null){
            return 0;
        }
        else{
            int left = dfs(root.left);
            int right = dfs(root.right);
            if(left > right){
                tileSum += left -right;
            }else{
                tileSum += right -left; 
            }
            return left + right + root.val;
        }
    }
}

File: go/563.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
var sum int

func findTilt(root *TreeNode) int {
	sum = 0
	dfs(root)
	return sum
}

func dfs(root *TreeNode) int {
	if root == nil {
		return 0
	}
	left := dfs(root.Left)
	right := dfs(root.Right)
	if left > right {
		sum += left - right
	} else {
		sum += right - left
	}
	return left + right + root.Val
}